kv/txn_coord_sender.go

```diff
@@ -709,12 +709,18 @@ func (tc *TxnCoordSender) updateState(ctx context.Context, ba roachpb.BatchReque
 		txnMeta := tc.txns[id]
 		// For successful transactional requests, keep the written intents and
 		// the updated transaction record to be sent along with the reply.
-		// The transaction metadata is created with the first writing operation
+		// The transaction metadata is created with the first writing operation.
+		// A tricky edge case is that of a transaction which "fails" on the
+		// first writing request, but actually manages to write some intents
+		// (for example, due to being multi-range). In this case, there will
+		// be an error, but the transaction will be marked as Writing.
```

Comment:
I think this comment needs more detail: why does it matter that the transaction is marked as Writing? I don't feel like i understand the issues here well enough to suggest some text.